api/v1alpha1/flowcollector_types.go

```diff
@@ -284,7 +284,7 @@ type FlowCollectorFLP struct {
 
 	// metricsServer endpoint configuration for Prometheus scraper
 	// +optional
-	MetricsServer MetricsServerConfig `json:"metricsServer,omitempty"`
+	MetricsServer *MetricsServerConfig `json:"metricsServer,omitempty"`
```

Comment:
Remember we can't rely on nil pointers for decisions due to the OLM console form.
What about a list enum that would also include the "enableKubeProbes" bool?

Something like:

```yaml
  monitoring: ['kube-probes', 'metrics']  
```

(though there's probably a better name than "monitoring" ..  )